lib/ueberauth/strategy/twitter.ex

```diff
@@ -0,0 +1,127 @@
+defmodule Ueberauth.Strategy.Twitter do
+  @moduledoc """
+  Twitter Strategy for Überauth.
+  """
+
+  use Ueberauth.Strategy, uid_field: :id_str
+
+  alias Ueberauth.Auth.Info
+  alias Ueberauth.Auth.Credentials
+  alias Ueberauth.Auth.Extra
+  alias Ueberauth.Strategy.Twitter
+
+  @doc """
+  Handles initial request for Twitter authentication.
+  """
+  def handle_request!(conn) do
+    token = Twitter.OAuth.request_token!([], [redirect_uri: callback_url(conn)])
+
+    conn
+    |> put_session(:twitter_token, token)
+    |> redirect!(Twitter.OAuth.authorize_url!(token))
+  end
+
+  @doc """
+  Handles the callback from Twitter.
+  """
+  def handle_callback!(%Plug.Conn{params: %{"oauth_verifier" => oauth_verifier}} = conn) do
+    token = get_session(conn, :twitter_token)
+    case Twitter.OAuth.access_token(token, oauth_verifier) do
+      {:ok, access_token} -> fetch_user(conn, access_token)
+      {:error, error} -> set_errors!(conn, [error(error.code, error.reason)])
+    end
+  end
+
+  @doc false
+  def handle_callback!(conn) do
+    set_errors!(conn, [error("missing_code", "No code received")])
+  end
+
+  @doc false
+  def handle_cleanup!(conn) do
+    conn
+    |> put_private(:twitter_user, nil)
+    |> put_private(:twitter_token, nil)
```

Comment:
Saw down below. I think this should also delete the twitter token from the session as well as the private assigns